### PR TITLE
chore: adding sentinel path for solomachine header

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

closes: #1915

### Commit / Changelog

chore(statemachine)!: Adding sentinel value for header path in 06-solomachine.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -69,7 +69,8 @@ Ref: https://keepachangelog.com/en/1.0.0/
 ### State Machine Breaking
 
 * (light-clients/07-tendermint) [\#2554](https://github.com/cosmos/ibc-go/pull/2554) Forbid negative values for `TrustingPeriod`, `UnbondingPeriod` and `MaxClockDrift` (as specified in ICS-07).
-* (06-solomachine) [\#2744](https://github.com/cosmos/ibc-go/pull/2744)  `Misbehaviour.ValidateBasic()` now only enforces that signature data does not match when the signature paths are different.
+* (06-solomachine) [\#2744](https://github.com/cosmos/ibc-go/pull/2744) `Misbehaviour.ValidateBasic()` now only enforces that signature data does not match when the signature paths are different.
+* (06-solomachine) [\#2748](https://github.com/cosmos/ibc-go/pull/2748) Adding sentinel value for header path in 06-solomachine.
```
Comment (chatton) [inline CHANGELOG.md:73]: let's add these to the description of the PR too in conventional format! ❤️

Comment (damiannolan) [reply CHANGELOG.md:73]: Is this correct?

`chore(statemachine)!: Adding sentinel value for header path in 06-solomachine.`

Comment (chatton) [reply CHANGELOG.md:73]: yeah that looks good!

File: modules/light-clients/06-solomachine/header.go [hunk 2]
```diff
@@ -10,6 +10,9 @@ import (
 	"github.com/cosmos/ibc-go/v6/modules/core/exported"
 )
 
+// SentinelHeaderPath defines a sentinel path used for solomachine header updates
```
Comment (damiannolan) [inline modules/light-clients/06-solomachine/header.go:13]:
```suggestion
// SentinelHeaderPath defines a placeholder path value used for headers in solomachine client updates
```

**Review by AdityaSripal — APPROVED** [review]

File: modules/light-clients/06-solomachine/header.go [hunk 3]
```diff
@@ -10,6 +10,9 @@ import (
 	"github.com/cosmos/ibc-go/v6/modules/core/exported"
 )
 
+// SentinelHeaderPath defines a placeholder path value used for headers in solomachine client updates
+const SentinelHeaderPath = "solomachine:header"
```
Comment (colin-axner) [inline modules/light-clients/06-solomachine/header.go:14]: should we match the proposed spec value? https://github.com/cosmos/ibc/pull/821/files#diff-028a9db43f8be864645baa5997fbb182b9af4cd9d444166944920c9fb25ee5fcR179

I'm okay proposing a different value in the spec as well

cc @AdityaSripal

**Review by colin-axner — APPROVED** [review]
ACK pending agreement on exact sentinel value

File: modules/light-clients/06-solomachine/header.go (hunk 3 above)
Comment (damiannolan) [reply modules/light-clients/06-solomachine/header.go:14]: Huh! I missed this was already part of the spec, my bad. 
I actually feel like proposing a change, what do you think? 

In my opinion `solomachine:header` feels a little easier on the eye than the capitalised snake case. Don't feel too strongly about it but just my two cents

Comment (colin-axner) [reply modules/light-clients/06-solomachine/header.go:14]: Sounds good to me!

**Review by chatton — APPROVED** [review]
LGTM (also agreeing with proposed sentinel value change!)

File: modules/light-clients/06-solomachine/header.go (hunk 3 above)
Comment (damiannolan) [reply modules/light-clients/06-solomachine/header.go:14]: https://github.com/cosmos/ibc/pull/821/files#r1025165561

**Pull request merged**